mgc/discrim/discrimTwoSample.py

```diff
@@ -0,0 +1,170 @@
+from sklearn.metrics import euclidean_distances
+from ._utils import _CheckInputs
+import numpy as np
+import random
+from .base import DiscriminabilityTest
+from scipy._lib._util import MapWrapper
+
+
+class DiscrimTwoSample(DiscriminabilityTest):
+    r"""
+     A class that performs a two-sample test for whether the discriminability is different for that of
+     one dataset vs another, as described in [#1Dscr]_.
+    
+    Parameters
+    ---------- 
+    is_dist : Boolean, optional (default: False)
+        whether `x1` and `x2` are distance matrices or not.
+    remove_isolates : Boolean, optional (default: True)
+        whether remove the samples with single instance or not.
+
+    See Also
+    --------
+    DiscrimOneSample : Onesample test for accessing discriminability of a single measurement
+    
+    Notes
+    -----
+    With :math:`\hat D_{X_1}` the sample discriminability of one approach, and
+    :math:`\hat D_{X_2}` the sample discriminability of another approach:
+    
+     .. math::
+
+         H_0: D_{X_1} = D_{X_2}
+
+    and
+
+     .. math::
+
+         H_A: D_{X_1} > D_{X_2}  
+    
+    Also implemented are tests of :math:`<` and :math:`\neq`.
+    """
+
+    def __init__(self, is_dist=False, remove_isolates=True):
+        self.is_distance = is_dist
+        self.remove_isolates = remove_isolates
+        DiscriminabilityTest.__init__(self)
+
+    def _statistic(self, x, y):
+        """
+        Helper function that calculates the discriminability test statistics.
+        """
+        stat_ = super(DiscrimTwoSample, self)._statistic(x, y)
+
+        return stat_
+
+    def test(self, x1, x2, y, reps=1000, alt="greater", workers=-1):
+        r"""
+        Calculates the test statistic and p-value for Discriminability two sample test.
+
+        Parameters
+        ----------
+        x1 : ndarray
+
+            * An `n \times d` data matrix with `n` samples in `d` dimensions. Should not be a distance matrix.  
```

Comment:
remove *

Reply:
change `n \times d` to `(n, d)`

Reply:
https://github.com/neurodata/mgc/blob/44ddaa1a8ed18edeeabd44418e4f1202ec27ddea/mgc/discrim/discrim_two_samp.py#L64-L67